persistence/src/main/scala/akka/persistence/SnapshotStoreInterop.scala

```diff
@@ -0,0 +1,108 @@
+package akka.persistence
+
+import akka.actor.ActorSystem
+import akka.persistence.SnapshotSelectionCriteria
+
+import cats.syntax.all._
+import cats.effect.Sync
+
+import com.evolutiongaming.catshelper.FromFuture
+import com.evolutiongaming.akkaeffect.ActorEffect
+import com.evolutiongaming.akkaeffect.persistence.{SnapshotStore, EventSourcedId, SeqNr}
+
+import scala.concurrent.duration._
+import java.time.Instant
+
+object SnapshotStoreInterop {
+
+  def apply[F[_]: Sync: FromFuture, A](
+    system: ActorSystem,
+    timeout: FiniteDuration,
+    snapshotPluginId: String,
+    eventSourcedId: EventSourcedId
+  ): F[SnapshotStore[F, A]] =
+    Sync[F]
+      .delay {
+        val actorRef = Persistence(system).snapshotStoreFor(snapshotPluginId)
+        ActorEffect.fromActor(actorRef)
+      }
+      .map { snapshotter =>
+        new SnapshotStore[F, A] {
+
+          val persistenceId = eventSourcedId.value
+
+          override def latest: F[Option[SnapshotStore.Offer[A]]] = {
+            val criteria = SnapshotSelectionCriteria()
+            val request  = SnapshotProtocol.LoadSnapshot(persistenceId, criteria, Long.MaxValue)
+            snapshotter
+              .ask(request, timeout)
+              .flatMap { response =>
+                response.flatMap {
+
+                  case SnapshotProtocol.LoadSnapshotResult(snapshot, _) =>
+                    snapshot match {
+
+                      case Some(offer) =>
+                        val payload   = offer.snapshot.asInstanceOf[A]
```

Comment:
Technically, this is impure operation, which may fail if snapshot of a different class is provided.

It might be worth to consider if it is acceptable or it is worth to hide it under `ApplicativeError[F].catchNonFatal` or `#castM` from `CatsHelper`,

Reply:
https://github.com/evolution-gaming/akka-effect/pull/276#discussion_r1450337030

Reply:
@dfakhritdinov In this case is is so cheap to be pedantic (with `MonadThrow[F].catchNonFatal`), so, may be you will reconsider?
```scala
                        val payload   = MonadThrow[F].catchNonFatal(offer.snapshot.asInstanceOf[A])
                        val timestamp = Instant.ofEpochMilli(offer.metadata.timestamp)
                        val metadata  = SnapshotStore.Metadata(offer.metadata.sequenceNr, timestamp)
                        payload.map(SnapshotStore.Offer(_, metadata).some)
```